src/main/java/uk/gov/hmcts/reform/pip/data/management/controllers/LocationMetadataController.java

```diff
@@ -0,0 +1,113 @@
+        this.locationMetadataService = locationMetadataService;
+    }
+
+    @PostMapping
+    @ApiResponse(responseCode = CREATED_CODE, description = "Add Location metadata")
+    @ApiResponse(responseCode = BAD_REQUEST_CODE, description = "Unable to add the location metadata")
+    @ApiResponse(responseCode = UNAUTHORISED_CODE, description = UNAUTHORISED_MESSAGE)
+    @ApiResponse(responseCode = FORBIDDEN_CODE, description = FORBIDDEN_MESSAGE)
+    @IsAdmin
+    @SecurityRequirement(name = BEARER_AUTHENTICATION)
+    public ResponseEntity<String> addLocationMetaData(@RequestHeader("x-requester-id") String userId,
+                                                      @RequestBody LocationMetadata locationMetadata) {
+        locationMetadataService.createLocationMetadata(locationMetadata, userId);
+        return ResponseEntity.status(HttpStatus.CREATED)
+            .body(String.format(
+                "Location metadata successfully added by user %s",
+                userId
+            ));
+    }
+
+    @PutMapping
+    @ApiResponse(responseCode = OK_CODE, description = "Update Location metadata")
+    @ApiResponse(responseCode = BAD_REQUEST_CODE, description = "Unable to update the location metadata")
+    @ApiResponse(responseCode = UNAUTHORISED_CODE, description = UNAUTHORISED_MESSAGE)
+    @ApiResponse(responseCode = FORBIDDEN_CODE, description = FORBIDDEN_MESSAGE)
+    @IsAdmin
+    @SecurityRequirement(name = BEARER_AUTHENTICATION)
+    public ResponseEntity<String> updateLocationMetaData(@RequestHeader("x-requester-id") String userId,
+                                                      @RequestBody LocationMetadata locationMetadata) {
+        locationMetadataService.updateLocationMetadata(locationMetadata, userId);
+        return ResponseEntity.status(HttpStatus.OK)
+            .body(String.format(
+                "Location metadata successfully updated by user %s",
+                userId
+            ));
+    }
+
+    @DeleteMapping("/{id}")
+    @ApiResponse(responseCode = OK_CODE, description = "Delete Location metadata")
+    @ApiResponse(responseCode = BAD_REQUEST_CODE, description = "Unable to delete the location metadata")
+    @ApiResponse(responseCode = UNAUTHORISED_CODE, description = UNAUTHORISED_MESSAGE)
+    @ApiResponse(responseCode = FORBIDDEN_CODE, description = FORBIDDEN_MESSAGE)
+    @IsAdmin
+    @SecurityRequirement(name = BEARER_AUTHENTICATION)
+    public ResponseEntity<String> deleteLocationMetaData(@RequestHeader("x-requester-id") String userId,
+                                                         @PathVariable String id) {
+        locationMetadataService.deleteById(id, userId);
+        return ResponseEntity.status(HttpStatus.OK)
+            .body(String.format(
+                "Location metadata successfully deleted by user %s",
+                userId
+            ));
+    }
+
+    @GetMapping("/{id}")
+    @ApiResponse(responseCode = OK_CODE, description = "Get Locations metadata")
```

Comment:
Can we change the API description on this to be Get Locations metadata by ID, or similar? Same with the one by location ID below

Reply:
done.